tests/ui/simd-abi-checks-s390x.rs

```diff
@@ -0,0 +1,174 @@
+//@ revisions: z10 z13_no_vector z13_soft_float
+//@ build-fail
+//@[z10] compile-flags: --target s390x-unknown-linux-gnu
+//@[z10] needs-llvm-components: systemz
+//@[z13_no_vector] compile-flags: --target s390x-unknown-linux-gnu -C target-cpu=z13 -C target-feature=-vector
+//@[z13_no_vector] needs-llvm-components: systemz
+// FIXME: +soft-float itself doesn't set -vector
+//@[z13_soft_float] compile-flags: --target s390x-unknown-linux-gnu -C target-cpu=z13 -C target-feature=-vector,+soft-float
```

Comment:
I added a soft-float case per https://github.com/rust-lang/rust/pull/131586#pullrequestreview-2365802446, but the soft-float target feature affects the float ABI, so its use with the -C target-feature flag will likely be deprecated (https://github.com/rust-lang/rust/issues/116344).

Also, the soft-float target feature is recognized by LLVM but not by rustc, so I get the following warning:
```
warning: unknown and unstable feature specified for `-Ctarget-feature`: `soft-float`
  |
  = note: it is still passed through to the codegen backend, but use of this feature might be unsound and the behavior of this feature can change in the future
```
https://godbolt.org/z/x75bPveh5

Reply:
(Btw, it seems other architecture's soft-float targets also explicitly disable SIMD-related features)

https://github.com/rust-lang/rust/blob/6faf0bd3e561f1a0c81f3eafe0ce0e688385d70e/compiler/rustc_target/src/spec/targets/x86_64_unknown_none.rs#L23

Reply:
Just saw this:

>but the soft-float target feature affects the float ABI, so its use with the -C target-feature flag will likely be deprecated

The only user of soft-float on our platform is the Linux kernel.  But given that Rust code is now added to the Linux kernel, it is crucial that it will remain possible to enable soft-float when compiling the kernel.   Of course, this affects the whole code base, not on a function-by-function basis -- I assume that will remain supported?

Reply:
> I assume that will remain supported?

Yes. What will be deprecated is the use of `-C target-feature` flag to override float ABI for targets defined as using hard-float ABI (or vice versa). This does not affect use cases that the target is defined as using soft-float ABI and you do not attempt to change its ABI.

Reply:
Maybe I'm still confused then.  Building the Linux kernel does not use another *target* as such.  It uses the regular s390x-linux target plus additional flags, some of which do change the ABI.  (Which is fine as they're used consistently across all files.)   While s390x support for Rust is not quite upstream yet, other platforms also use ABI-changing options when compiling Rust kernel code today ...

Reply:
AFAIK, Rust's `*-vecdor-linux-{gnu,musl,..}` targets are for userland, and Rust for Linux uses bare-metal soft-float targets to compile kernel code: https://github.com/rust-lang/rust/pull/129884#issuecomment-2325047544